controllers/comments_controller.js

```diff
@@ -0,0 +1,30 @@
+const fs = require("fs/promises");
+const { selectComments, insertComments } = require("../models/comments_model");
+const { selectArticleById } = require("../models/articles_model");
+
+exports.getCommentsByArticleId = (req, res, next) => {
+  const { article_id } = req.params;
+  const promises = [selectArticleById(article_id), selectComments(article_id)];
+  Promise.all(promises)
+    .then((resultPromises) => {
+      res.status(200).send({ comments: resultPromises[1] });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+exports.postComment = (req, res, next) => {
+  const { article_id } = req.params;
+  const promises = [
+    selectArticleById(article_id),
```

Comment:
This method of checking an article exists is fine, but is not needed here!   Article_id is also a foreign key on comments table so has to exist.   A separate check that the article exists should not be necessary as a psql error will be thrown instead.  This psql may have a code you are not catching though.  So this may need updating.